include/_antic.h

```diff
@@ -57,42 +57,42 @@ struct __antic {
-#define DL_CHR40x8x4    4       /* colour, 40 character & 8 scanlines per mode line (GR. 12) */
-#define DL_CHR40x16x4   5       /* colour, 40 character & 16 scanlines per mode line (GR. 13) */
-#define DL_CHR20x8x2    6       /* colour (duochrome per character), 20 character & 8 scanlines per mode line (GR. 1) */
-#define DL_CHR20x16x2   7       /* colour (duochrome per character), 20 character & 16 scanlines per mode line (GR. 2) */
+#define DL_CHR40x8x1    (unsigned char) 2       /* monochrome, 40 character & 8 scanlines per mode line (GR. 0) */
+#define DL_CHR40x10x1   (unsigned char) 3       /* monochrome, 40 character & 10 scanlines per mode line */
+#define DL_CHR40x8x4    (unsigned char) 4       /* colour, 40 character & 8 scanlines per mode line (GR. 12) */
+#define DL_CHR40x16x4   (unsigned char) 5       /* colour, 40 character & 16 scanlines per mode line (GR. 13) */
+#define DL_CHR20x8x2    (unsigned char) 6       /* colour (duochrome per character), 20 character & 8 scanlines per mode line (GR. 1) */
+#define DL_CHR20x16x2   (unsigned char) 7       /* colour (duochrome per character), 20 character & 16 scanlines per mode line (GR. 2) */
 
-#define DL_MAP40x8x4    8       /* colour, 40 pixel & 8 scanlines per mode line (GR. 3) */
-#define DL_MAP80x4x2    9       /* 'duochrome', 80 pixel & 4 scanlines per mode line (GR.4) */
-#define DL_MAP80x4x4    10      /* colour, 80 pixel & 4 scanlines per mode line (GR.5) */
-#define DL_MAP160x2x2   11      /* 'duochrome', 160 pixel & 2 scanlines per mode line (GR.6) */
-#define DL_MAP160x1x2   12      /* 'duochrome', 160 pixel & 1 scanline per mode line (GR.14) */
-#define DL_MAP160x2x4   13      /* 4 colours, 160 pixel & 2 scanlines per mode line (GR.7) */
-#define DL_MAP160x1x4   14      /* 4 colours, 160 pixel & 1 scanline per mode line (GR.15) */
-#define DL_MAP320x1x1   15      /* monochrome, 320 pixel & 1 scanline per mode line (GR.8) */
+#define DL_MAP40x8x4    (unsigned char) 8       /* colour, 40 pixel & 8 scanlines per mode line (GR. 3) */
+#define DL_MAP80x4x2    (unsigned char) 9       /* 'duochrome', 80 pixel & 4 scanlines per mode line (GR.4) */
+#define DL_MAP80x4x4    (unsigned char) 10      /* colour, 80 pixel & 4 scanlines per mode line (GR.5) */
+#define DL_MAP160x2x2   (unsigned char) 11      /* 'duochrome', 160 pixel & 2 scanlines per mode line (GR.6) */
+#define DL_MAP160x1x2   (unsigned char) 12      /* 'duochrome', 160 pixel & 1 scanline per mode line (GR.14) */
+#define DL_MAP160x2x4   (unsigned char) 13      /* 4 colours, 160 pixel & 2 scanlines per mode line (GR.7) */
+#define DL_MAP160x1x4   (unsigned char) 14      /* 4 colours, 160 pixel & 1 scanline per mode line (GR.15) */
+#define DL_MAP320x1x1   (unsigned char) 15      /* monochrome, 320 pixel & 1 scanline per mode line (GR.8) */
 
 /* modifiers on mode lines */
-#define DL_HSCROL       16
-#define DL_VSCROL       32
-#define DL_LMS          64
+#define DL_HSCROL(x)    (unsigned char)((x) | 16)
```

Comment:
> I really would appreciate a single definition as two concurrent ones can be confusing, mixed and misleading. I see no real benefit in having different ones.

Backwards compatibility.

> but thanks to the new definition you aren't in danger to use a "wrong operator"...

When writing C you're always in danger of using a wrong operator.

> What about a macro for that use case?

That would be nice.

> DisplayList[2] |= DL_DLI(0)

So, it boils down to having the same name or a different name for the new DL_DLI(x) macro with the existing DL_DLI macro (and of course the other modifiers), correct?

My POV is that if there is no real good reason to break backward compatibilty, don't do it. I don't think this case is a real good reason. As you stated before, previously there was no documentation for the Antic modes, so it could be documented that there is DL_DLI for the constant, and (say) DL_DLIx() for the modifier macro.
cc65 already seems to be difficult to newbies, so they tend to shy away, and if there are complaints that something that worked before suddenly doesn't work anymoore with the newest version will raise the acceptance level even more.